admiral/pkg/clusters/admiralDatabaseClient.go

```diff
@@ -93,13 +136,90 @@ func NewAdmiralDatabaseClient(admiralConfigPath string, dynamoClientInitFunc fun
 		return nil, fmt.Errorf("error unmarshalling admiral config file, err: %v", err)
 	}
 
-	workloadDatabaseClient.database = &admiralConfig.WorkloadDatabase
-	workloadDatabaseClient.dynamoClient, err = dynamoClientInitFunc(
+	return admiralConfig, nil
+}
+
+func NewDynamicConfigDatabaseClient(path string, dynamoClientInitFunc func(role string, region string) (*DynamoClient, error)) (*DynamicConfigDatabaseClient, error) {
+	var dynamicConfigClient = DynamicConfigDatabaseClient{}
+
+	admiralConfig, err := ReadDynamoConfigForDynamoDB(path)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling admiral config file, err: %v", err)
+	}
+
+	dynamicConfigClient.database = &admiralConfig.DynamicConfigDatabase
+	dynamicConfigClient.database.TableName = common.GetAdmiralParams().DynamicConfigDynamoDBTableName
+	dynamicConfigClient.dynamoClient, err = dynamoClientInitFunc(
 		admiralConfig.WorkloadDatabase.Role,
 		admiralConfig.WorkloadDatabase.Region,
 	)
 	if err != nil {
-		return nil, fmt.Errorf("unable to instantiate dynamo client, err: %v", err)
+		return &dynamicConfigClient, fmt.Errorf("unable to instantiate dynamo client for DynamicConfig, err: %v", err)
+	}
+	return &dynamicConfigClient, nil
+}
+
+func UpdateASyncAdmiralConfig(dbClient AdmiralDatabaseManager, syncTime int) {
+
+	for range time.Tick(time.Minute * time.Duration(syncTime)) {
+		ReadAndUpdateSyncAdmiralConfig(dbClient)
+	}
+}
+
+func ReadAndUpdateSyncAdmiralConfig(dbClient AdmiralDatabaseManager) error {
+
+	dbRawData, err := dbClient.Get("EnableDynamicConfig", common.Admiral)
+	if err != nil {
+		log.Errorf("Error getting EnableDynamicConfig admiral config, err: %v", err)
+		return err
+	}
+
+	configData, ok := dbRawData.(DynamicConfigData)
+	if !ok {
+		return errors.New("Failed to parse DynamicConfigData")
+	}
+
+	if IsDynamicConfigChanged(configData) {
+		log.Infof("Updating DynamicConfigData with Admiral config")
+		UpdateSyncAdmiralConfig(configData)
+	} else {
+		log.Infof("No need to update DynamicConfigData")
+	}
+
+	return nil
+}
+
+func IsDynamicConfigChanged(config DynamicConfigData) bool {
+
+	if config.EnableDynamicConfig != common.Admiral {
+		return false
+	}
+
+	if DynamicConfigCheckSum == sha256.Sum256([]byte(fmt.Sprintf("%v", config))) {
```

Comment:
nit pick, we had checked m5s vs sha256 for speed and cpu, for a different requirement and chose md5 for being a little fast and consuming a little less CPU

```
$ openssl speed md5 sha256
The 'numbers' are in 1000s of bytes per second processed.
type             16 bytes     64 bytes    256 bytes   1024 bytes   8192 bytes  16384 bytes
md5              51896.59k   156143.77k   336557.91k   472481.13k   536428.54k   543648.43k
sha256           42111.47k   120368.74k   247776.17k   335801.69k   376607.34k   377296.21k
```